app/views/portal/show.rb

```diff
@@ -0,0 +1,871 @@
+              fields: ['proxies.dctermsIsRequiredBy']
+            },
+            {
+              title: 'site.object.meta-label.edm.has-met',
+              fields: ['proxies.edmHasMet']
+            },
+            {
+              title: 'site.object.meta-label.edm.incorporates',
+              fields: ['proxies.edmIncorporates']
+            },
+            {
+              title: 'site.object.meta-label.edm.is-derivative-of',
+              fields: ['proxies.edmIsDerivativeOf']
+            },
+            {
+              title: 'site.object.meta-label.edm.is-representation-of',
+              fields: ['proxies.edmIsRepresentationOf']
+            },
+            {
+              title: 'site.object.meta-label.edm.is-similar-to',
+              fields: ['proxies.edmIsSimilarTo']
+            },
+            {
+              title: 'site.object.meta-label.edm.is-successor-of',
+              fields: ['proxies.edmIsSuccessorOf']
+            },
+            {
+              title: 'site.object.meta-label.edm.realises',
+              fields: ['proxies.edmRealizes']
+            },
+            {
+              title: 'site.object.meta-label.edm.was-present-at',
+              fields: ['proxies.edmRealizes']
+            }
+          ]
+        ),
+        similar: {
+          title: t('site.object.similar-items') + ':',
+          more_items_query: search_path(mlt: document.id),
+          items: @similar.map do |doc|
+            {
+              url: document_path(doc, format: 'html'),
+              title: render_document_show_field_value(doc, ['dcTitleLangAware', 'title']),
+              img: {
+                alt: render_document_show_field_value(doc, ['dcTitleLangAware', 'title']),
+                src: render_document_show_field_value(doc, 'edmPreview')
+              }
+            }
+          end
+        },
+        thumbnail: render_document_show_field_value(document, 'europeanaAggregation.edmPreview', tag: false)
+      }
+    end
+
+    def labels
+      {
+        show_more_meta: t('site.object.actions.show-more-data'),
+        show_less_meta: t('site.object.actions.show-less-data'),
+        rights: t('site.object.meta-label.rights-human')
+      }
+    end
+
+    private
+
+    def edm_is_shown_by_download_url
+      @edm_is_shown_by_download_url ||= begin
+        if ENV['EDM_IS_SHOWN_BY_PROXY'] && document.aggregations.size > 0 && document.aggregations.first.fetch('edmIsShownBy', false)
+          ENV['EDM_IS_SHOWN_BY_PROXY'] + document.fetch('about', '/')
+        else
+          render_document_show_field_value(document, 'aggregations.edmIsShownBy')
+        end
+      end
+    end
+
+    def collect_values(fields, doc = document)
+      fields.map do |field|
+        render_document_show_field_value(doc, field)
+      end.compact.uniq
+    end
+
+    def merge_values(fields, separator = ' ')
+      collect_values(fields).join(separator)
+    end
+
+    def data_section(data)
```

Comment:
Cyclomatic complexity for data_section is too high. [29/6]<br>Assignment Branch Condition size for data_section is too high. [103.6/15]<br>Method has too many lines. [108/10]<br>Perceived complexity for data_section is too high. [32/7]